Source/WebKit/NetworkProcess/cache/NetworkCache.cpp

```diff
@@ -284,6 +284,12 @@ static StoreDecision makeStoreDecision(const WebCore::ResourceRequest& originalR
             return StoreDecision::NoDueToHTTPStatusCode;
     }
 
+    // FIXME: Our caching does not take into account fragments properly.
+    // If we cache localhost/#key=foo we will return the same parameters key=foo
+    // on the next visit to the domain even if the parameters have been updated.
+    if ((response.isRedirection() || response.isRedirected()) && originalRequest.urlContainsFragmentIdentifier())
```

Comment:
Not sure why we need `response.isRedirection()`

Reply:
This was handling loading the redirected page.
This might not be needed.